src/gfn/states.py

```diff
@@ -850,12 +852,17 @@ def extend(self, other: GraphStates):
             ],
             dim=0,
         )
-        assert torch.all(
-            self.tensor["batch_shape"][1:] == other.tensor["batch_shape"][1:]
-        )
-        self.tensor["batch_shape"] = (
-            self.tensor["batch_shape"][0] + other.tensor["batch_shape"][0],
-        ) + self.batch_shape[1:]
+
+        #self.tensor["batch_shape"] = self.tensor["batch_shape"] + other.tensor["batch_shape"]
+        # If self.tensor is a placeholder and all batch_dims are 0, this check won't pass.
+        if not torch.all(self.tensor["batch_shape"] == 0):
+            assert torch.all(
+                self.tensor["batch_shape"][1:] == other.tensor["batch_shape"][1:]
+            )
+        # self.tensor["batch_shape"] = (
+        #     self.tensor["batch_shape"][0] + other.tensor["batch_shape"][0],
+        # ) + self.batch_shape[1:]
+        self.tensor["batch_shape"] = self.tensor["batch_shape"] + other.tensor["batch_shape"]
```

Comment:
The problem is with these lines.

When you extend a tensor, you should only sum the first dimension.
Imagine with tensors, if you do
(10, 200) . extends (30, 200) you should get (40, 200) not (40, 400)

However, this will throw an error if you try to extend (0, 0) with (30, 200) because 0 != 200, but I believe this is expected and should be changed above in the stack trace

Reply:
This is resolved (with the move to PYG).